apis/ec2/v1beta1/natgateway_types.go

```diff
@@ -0,0 +1,126 @@
+
+	// SubnetIDRef references a subnet and retrives it's subnet id
+	// +immutable
+	// +optional
+	SubnetIDRef *runtimev1alpha1.Reference `json:"subnetIdRef,omitempty"`
+
+	// SubnetIDSelector references a subnet by selector and retrives it's subnet id
+	// +immutable
+	// +optional
+	SubnetIDSelector *runtimev1alpha1.Selector `json:"subnetIdSelector,omitempty"`
+
+	// Tags represents to current ec2 tags.
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+}
+
+// NatGatewaySpec defines the desired state of a NAT Gateway
+type NatGatewaySpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  NatGatewayParameters `json:"forProvider"`
+}
+
+// NatGatewayObservation keeps the state for the CR
+type NatGatewayObservation struct {
+	CreateTime          *metav1.Time        `json:"createTime,omitempty"`
+	DeleteTime          *metav1.Time        `json:"deleteTime,omitempty"`
+	FailureCode         string              `json:"failureCode,omitempty"`
+	FailureMessage      string              `json:"failureMessage,omitempty"`
+	NatGatewayAddresses []NatGatewayAddress `json:"natGatewayAddresses,omitempty"`
+	NatGatewayID        string              `json:"natGatewayId"`
+	State               string              `json:"state,omitempty"`
+	SubnetID            string              `json:"subnetId"`
+	Tags                []Tag               `json:"tags,omitempty"`
+	VpcID               string              `json:"vpcId"`
+}
+
+// NatGatewayAddress describes the details of network
+type NatGatewayAddress struct {
+	AllocationID       string `json:"allocationId"`
+	NetworkInterfaceID string `json:"networkInterfaceId"`
+	PrivateIP          string `json:"privateIp"`
+	PublicIP           string `json:"publicIp"`
+}
+
+// NatGatewayStatus describes the observed state
+type NatGatewayStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     NatGatewayObservation `json:"atProvider"`
+}
+
+// +kubebuilder:object:root=true
+
+// An NatGateway is a managed resource that represents an AWS VPC NAT
+// Gateway.
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="ID",type="string",JSONPath=".metadata.annotations.crossplane\\.io/external-name"
+// +kubebuilder:printcolumn:name="VPC",type="string",JSONPath=".status.atProvider.vpcId"
+// +kubebuilder:printcolumn:name="SUBNET",type="string",JSONPath=".spec.forProvider.subnetId"
+// +kubebuilder:printcolumn:name="ALLOCATION ID",type="string",JSONPath=".spec.forProvider.allocationId"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,aws}
+type NatGateway struct {
```

Comment:
```suggestion
type NATGateway struct {
```

We have been following the convention of keeping abbreviations upper-case.

Reply:
fixed